modelator/src/util.rs

```diff
@@ -79,3 +81,65 @@ pub(crate) mod digest {
         Ok(())
     }
 }
+
+/// Copies all files with the given extension in the same directory as the given file into another directory
+/// Returns the new path for the main file
+pub(crate) fn copy_files_into<P: AsRef<Path>, Q: AsRef<Path>>(
+    ext: &str,
+    file: P,
+    dir: Q,
+) -> Result<PathBuf, Error> {
+    let files = list_files(ext, &file)?;
+    let dir = PathBuf::from(dir.as_ref());
+    if !dir.is_dir() || !dir.exists() {
+        return Err(Error::IO(
+            "Can't copy files: destination directory doesn't exist".to_string(),
+        ));
+    }
+    for file in files {
+        if let Some(file_name) = file.file_name() {
+            let dest = dir.join(file_name);
+            copy(file, dest).map_err(Error::io)?;
+        }
+    }
+    // it is OK to unwrap, as the file has been checked before
+    Ok(dir.join(file.as_ref().file_name().unwrap()))
+}
+
+/// Lists all files with the given extension in the same directory as the given file
+/// Returns error if the given file doesn't exist, or is not a TLA+ file
+fn list_files<P: AsRef<Path>>(ext: &str, file: P) -> Result<Vec<PathBuf>, Error> {
+    // Checks that the given path points to an existing TLA+ file
+    let is_ext = |f: &Path| {
+        f.extension()
+            .map_or("".to_owned(), |x| x.to_string_lossy().to_string())
+            == ext
+    };
+    let file = file.as_ref();
+    if !file.exists() || !is_ext(file) {
+        return Err(Error::IO(format!("File doesn't exist: {}", file.display())));
+    }
+    // The parent directory of the file
+    let dir = file.parent().map_or(PathBuf::from("./"), |p| {
+        if p.to_string_lossy().is_empty() {
+            PathBuf::from("./")
+        } else {
+            p.to_path_buf()
+        }
+    });
+    // Collect all files with the same extension in this directory
+    let mut files = Vec::new();
+    for entry in WalkDir::new(dir)
+        .min_depth(1)
+        .max_depth(1)
+        .into_iter()
```

Comment:
Do we really need `walkdir` for this? `std::fs::read_dir` should be enough, no?

Reply:
Sure! How about you refactor this to use `std::fs::read_dir`, right in this branch? it would be great to avoid an additional dependency, true.

Reply:
Done! Ran `cargo test`.

Reply:
Great, looks good!thanks:)